pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3037,3 +3046,430 @@ func TestUpdatePipelineRunStatusFromTaskRuns(t *testing.T) {
 		})
 	}
 }
+
+// this test validates pipeline with finally for various test cases including:
+// Test 01 - Pipeline run should result in error when a dag task fails but final task is executed successfully.
+// Test 02 - Pipeline run should result in error when a dag task is successful but final task fails.
+// Test 03 - Pipeline run should result in error when both dag task and final task fail.
+// Test 04 - Pipeline run should not schedule final tasks while dag tasks are still running.
+// Test 05 - Pipeline run should not schedule final tasks while dag tasks are still running and no other dag task available to schedule.
```

Comment:
Those can be removed, they are in the name of the test struct

Reply:
Done